src/main/java/org/mule/extension/file/common/api/matcher/PathMatcherPredicate.java

```diff
@@ -30,13 +29,18 @@ public final class PathMatcherPredicate implements Predicate<String> {
 
   private final Predicate<String> delegate;
 
+
+
   /**
    * Creates a new instance using the given pattern
    *
    * @param pattern the pattern to be used to test paths.
+   * @param predicateType if is gonna a match local file system or a remote file system ex:ftp , sftp
+   * @param caseSensitive if the predicateType is a external file system predicate this set the case sensitivity
    */
-  public PathMatcherPredicate(String pattern) {
-    delegate = getPredicateForFilename(pattern);
+
+  public PathMatcherPredicate(String pattern, PredicateType predicateType, final boolean caseSensitive) {
```

Comment:
There is something wrong here. You cannot change the public methods or constructors (mainly when you are working on an API package).
Could you preserve the old signature and overload it with the new parameters?
(Please take into account that the default values should be consistent between classes)

Reply:
Yes sure